cmd/kubelet/app/options/options.go

```diff
@@ -363,6 +362,8 @@ func (f *KubeletFlags) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&f.BootstrapCheckpointPath, "bootstrap-checkpoint-path", f.BootstrapCheckpointPath, "<Warning: Alpha feature> Path to to the directory where the checkpoints are stored")
 
 	// DEPRECATED FLAGS
+	fs.Int32Var(&f.CAdvisorPort, "cadvisor-port", f.CAdvisorPort, "The port of the localhost cAdvisor endpoint (set to 0 to disable)")
+	fs.MarkDeprecated("cadvisor-port", "The default will change to 0 (disabled) in 1.11, and the cadvisor port will be removed entirely in 1.13")
```

Comment:
Should we give 6 months of warning (v1.12) before disabling? This would fit more with the [deprecation policy](https://kubernetes.io/docs/reference/deprecation-policy/), which specifies the longer of 6 months or 1 release, given our current quarterly release cadence. Since disabling has the same effect as deprecation (the server disappears) for anyone who doesn't explicitly set this, it seems like v1.12 might be better than v1.11.

Reply:
I thought the timeline didn't apply to defaults, just the presence of the flag.  Although I don't think its a big deal to move it back a release.  Does this mean all changes to flag defaults need 6 mo warning?

Reply:
Changes to defaults are API changes, and should follow the deprecation policy's warning timelines. If this were a declarative API, rather than a CLI, changes to defaults would require incrementing the API version.

Reply:
changed to 1.12